### PR TITLE
Add --service-lb-annotation flag to install command

### PR DESCRIPTION
**Pull request by tylerslaton**
This allows you to configure Acorn to add annotations required by external cluster LoadBalancer controllers (like the `aws-load-balancer-controller`). You can set them like so:

```sh
acorn install --service-lb-annotations foo=bar
```

You can specify multiple by comma delimiting them

```sh
acorn install --service-lb-annotations foo=bar,baz=bat
```

If you'd like to unset them, you can do so by providing an empty string.

```sh
acorn install --service-lb-annotations ""
```

### Checklist
- [X] The title of this PR would make a good line in Acorn's Release Note's Changelog
- [X] The title of this PR ends with a link to the main issue being address in parentheses, like: `This is a title (#1216)`. [Here's an example](https://github.com/acorn-io/acorn/pull/1199)
- [X] All relevant issues are referenced in the PR description. *NOTE: don't use [GitHub keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) that auto-close issues*
- [X] Commits follow [contributing guidance](https://github.com/acorn-io/acorn/blob/main/CONTRIBUTING.md#commits)
- [ ] Automated tests added to cover the changes. If tests couldn't be added, an explanation is provided in the Verification and Testing section
- [X] Changes to user-facing functionality, API, CLI, and upgrade impacts are clearly called out in PR description
- [ ] PR has at least two approvals before merging (or a reasonable exception, like it's just a docs change)

### CODE REVIEW

File: pkg/config/config.go [hunk 1]
```diff
@@ -316,6 +316,12 @@ func merge(oldConfig, newConfig *apiv1.Config) *apiv1.Config {
 		mergedConfig.PropagateProjectLabels = newConfig.PropagateProjectLabels
 	}
 
+	if len(newConfig.ServiceLBAnnotations) == 0 {
+		mergedConfig.ServiceLBAnnotations = nil
+	} else if len(newConfig.ServiceLBAnnotations) > 0 {
```
Comment (thedadams) [inline pkg/config/config.go:321]: nit:
```suggestion
	} else {
```

File: pkg/publish/servicelb.go [hunk 2]
```diff
@@ -33,6 +35,21 @@ func ServiceLoadBalancer(svc *v1.ServiceInstance) (result []kclient.Object, _ er
 		return nil, nil
 	}
 
+	cfg, err := config.Get(req.Ctx, req.Client)
+	if err != nil {
+		return nil, err
+	}
+
+	if cfg.ServiceLBAnnotations != nil && len(cfg.ServiceLBAnnotations) > 0 {
```
Comment (thedadams) [inline pkg/publish/servicelb.go:43]: This check is not necessary. If there is nothing to range over, then nothing will happen.

File: pkg/publish/servicelb.go [hunk 3]
```diff
@@ -33,6 +35,21 @@ func ServiceLoadBalancer(svc *v1.ServiceInstance) (result []kclient.Object, _ er
...
+		if svc.Spec.Annotations == nil {
+			svc.Spec.Annotations = map[string]string{}
+		}
+
+		for k, v := range cfg.ServiceLBAnnotations {
```
Comment (thedadams) [inline pkg/publish/servicelb.go:48]: Let's sort these with `typed.Sorted` to ensure the output is consistent.

File: pkg/apis/api.acorn.io/v1/types.go [hunk 4]
```diff
@@ -336,34 +336,35 @@ type Config struct {
...
+	ManageVolumeClasses            *bool             `json:"manageVolumeClasses" name:"manage-volume-classes" usage:"Manually manage volume classes rather than sync with storage classes, setting to 'true' will delete Acorn-created volume classes"`
+	NetworkPolicies                *bool             `json:"networkPolicies" name:"network-policies" usage:"Create Kubernetes NetworkPolicies which block cross-project network traffic (default true)"`
+	IngressControllerNamespace     *string           `json:"ingressControllerNamespace" name:"ingress-controller-namespace" usage:"The namespace where the ingress controller runs - used to secure published HTTP ports with NetworkPolicies."`
+	AllowTrafficFromNamespace      []string          `json:"allowTrafficFromNamespace" name:"allow-traffic-from-namespace" usage:"Namespaces that are allowed to send network traffic to all Acorn apps"`
+	ServiceLBAnnotations           map[string]string `json:"serviceLBAnnotations" name:"service-lb-annotations" usage:"Annotations to add to the service of type LoadBalancer (defaults to empty)"`
```
Comment (thedadams) [inline pkg/apis/api.acorn.io/v1/types.go:367]: What is an example of how a user would provide these strings? Is it `--service-lb-annotations key=value`?

File: pkg/publish/servicelb.go (hunk 2 above)
Comment (tylerslaton) [reply pkg/publish/servicelb.go:43]: Makes sense, good catch.

File: pkg/publish/servicelb.go [hunk 5]
```diff
@@ -33,6 +36,18 @@ func ServiceLoadBalancer(svc *v1.ServiceInstance) (result []kclient.Object, _ er
...
 
+	cfg, err := config.Get(req.Ctx, req.Client)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, entry := range typed.Sorted(cfg.ServiceLBAnnotations) {
+		if svc.Spec.Annotations == nil {
+			svc.Spec.Annotations = map[string]string{}
+		}
```
Comment (thedadams) [inline pkg/publish/servicelb.go:47]: Can this check be pulled out of the for-loop?

Comment (tylerslaton) [reply pkg/publish/servicelb.go:47]: Yeah totally, I was a little nervous about setting it to empty when it was nil but thinking on it more its likely a no-op.

**Review by thedadams — APPROVED** [review]

**Review by cjellick — APPROVED** [review]

**Pull request merged**